src/debug/ee/debugger.cpp

```diff
@@ -15306,6 +15306,13 @@ HRESULT Debugger::FuncEvalSetup(DebuggerIPCE_FuncEvalInfo *pEvalInfo,
         return CORDBG_E_ILLEGAL_AT_GC_UNSAFE_POINT;
     }
 
+    if (::GetSP(filterContext) != ALIGN_DOWN(::GetSP(filterContext), STACK_ALIGN_SIZE))
+    {
+        // SP is not aligned, we cannot do a FuncEval here
+        LOG((LF_CORDB, LL_INFO1000, "D::FES SP is unaligned"));
```

Comment:
Do you know if the x86 JIT will always try to keep the stack alignment at 4-bytes? Since the processor isn't very picky, I wasn't sure if the JIT would try very hard for that.

Reply:
The x86 JIT should enforce the `STACK_ALIGN_SIZE`, the JIT uses `STACK_ALIGN`.  I looked at PR history, and there are fixes to fix x86 stack alignment specifically for Linux, but the JIT should enforce either Windows or Linux alignment correctly.